### DISCUSSION TITLE
Drag and Drop

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#6218] [Q&A]
Hi all,

I'm trying to implement a drag and drop solution with an adorner to show the listboxitem being dragged between two different listboxes. To do this, I'm using events from the usercontrol that contains both listboxes:

- pointerpressed event - if over a listboxitem, removes that item from the listbox and adds it to the adorner layer
- pointermoved event - This calls a method which updates the position of the listboxitem in the adornerlayer using a translate transform

So far so good and the dragging the item around is working nicely.

So now I want to make this into a proper drag and drop operation so I call await DragDrop.DoDragDrop() from the PointerMoved event just after the method that moves the adorner layer listboxitem. Now, I get the drag effects etc. on the mouse pointer indicating where I can drop but the listboxitem doesn't move any more. On closer inspection, the pointerover event is not firing when the mouse is moved while executing a dragdrop operation.

I then tried calling the adorner listboxitem moving method from the dragover event instead of the pointermoved event - this works to a point - as long as the mouse is over a control that has AllowDrop set, the adorner listboxitem's position is updated. However, when the mouse is over whitespace (such as the stackpanel containing the two listboxes), the dragover event doesn't fire and so the adorner listboxitem's position is not updated.

Can anyone suggest where I might be going wrong with this or if there is another approach I can try?

Thanks so much!
Chris

**Comment by dolejska-daniel** [comment]
Hi @renrutsirhc, have you learned anything more about this? Were you able to achieve what you set out to do in the end?

**Reply by timunie** [reply]
Check out https://github.com/wieslawsoltes/Xaml.Behaviors
